core/src/main/java/com/linecorp/armeria/server/ServiceRequestContext.java

```diff
@@ -543,4 +544,49 @@ default void timeoutNow() {
      * Returns the proxied addresses of the current {@link Request}.
      */
     ProxiedAddresses proxiedAddresses();
+
+    /**
+     * Initiates graceful connection shutdown with a given drain duration in microseconds.
+     *
+     * <p>
+     * At the connection drain server signals the clients that the connection shutdown is imminent
+     * but still accepts in flight requests.
+     * At the connection drain end server stops accepting new requests.
+     * </p>
+     *
+     * <p>
+     * If graceful shutdown is already triggered and the given connection drain duration is smaller than
+     * the wait time before the connection drain end - reschedules the drain end to happen faster.
+     * Otherwise, drain will end as it was previously scheduled.
+     * </p>
+     *
+     * <p>
+     * Note that HTTP/1 doesn't support draining as described here, so for HTTP/1 drain duration
+     * is always {@code 0}.
+     * </p>
+     *
+     * <p>
+     * Returns {@link CompletableFuture} that completes when the channel is closed.
```

Comment:
Could we merge this sentence and the first sentence so that a user knows what will be returned from the first sentence? e.g. `Initiates... and returns ...`